cross/lxc/patches/007-fix-apparmor-profiles.patch

```diff
@@ -0,0 +1,39 @@
+diff -ruN config/apparmor/abstractions/start-container.in config/apparmor.new/abstractions/start-container.in
+--- config/apparmor/abstractions/start-container.in	2021-01-12 00:20:05.000000000 +0000
++++ config/apparmor.new/abstractions/start-container.in	2021-04-12 23:03:28.503948847 +0000
+@@ -9,8 +9,8 @@
+   ptrace,
+ 
+   # currently blocked by apparmor bug
+-  mount -> /usr/lib*/*/lxc/{**,},
+-  mount -> /usr/lib*/lxc/{**,},
++  mount -> @PREFIX@/lib*/*/lxc/{**,},
++  mount -> @PREFIX@/lib*/lxc/{**,},
+   mount -> @LXCROOTFSMOUNT@/{,**},
+   mount fstype=devpts -> /dev/pts/,
+   mount options=bind /dev/pts/ptmx/ -> /dev/ptmx/,
+@@ -38,10 +38,10 @@
+   # This may look a bit redundant, however it appears we need all of
+   # them if we want things to work properly on all combinations of kernel
+   # and userspace parser...
+-  pivot_root /usr/lib*/lxc/,
+-  pivot_root /usr/lib*/*/lxc/,
+-  pivot_root /usr/lib*/lxc/**,
+-  pivot_root /usr/lib*/*/lxc/**,
++  pivot_root @PREFIX@/lib*/lxc/,
++  pivot_root @PREFIX@/lib*/*/lxc/,
++  pivot_root @PREFIX@/lib*/lxc/**,
++  pivot_root @PREFIX@/lib*/*/lxc/**,
+   pivot_root @LXCROOTFSMOUNT@/{,**},
+ 
+   change_profile -> lxc-*,
+diff -ruN config/apparmor/usr.bin.lxc-start config/apparmor.new/usr.bin.lxc-start
+--- config/apparmor/usr.bin.lxc-start	2021-01-12 00:20:05.000000000 +0000
++++ config/apparmor.new/usr.bin.lxc-start	2021-04-12 23:20:09.480000000 +0000
+@@ -1,5 +1,5 @@
+ #include <tunables/global>
+ 
+-/usr/bin/lxc-start flags=(attach_disconnected) {
++/var/packages/lxc/target/bin/lxc-start flags=(attach_disconnected) {
```

Comment:
Presuming this could use `@PREFIX@/bin` as well?

Reply:
I had a go but could not get it to work. I do not know enough right now about how autoconf works. I tried renaming the file to _\<filename\>_.**in** and inserted `@PREFIX@/bin` but autoconf didn't pick it up. I guess I am meant to have told autoconf about it somewhere else.

I've left this on my "figure it out later" list for now as I haven't got around to actually applying the apparmor profiles and checking they work yet.